WasatchNET/Spectrometer.cs

```diff
@@ -2119,28 +2119,28 @@ uint[] readImage(UsbEndpointReader spectralReader, int pixelsPerEndpoint)
                 try
                 {
                     int bytesToRead = chunk_size - bytesReadThisEndpoint;
-                    logger.debug("readSubspectrum: attempting to read {0} bytes of spectrum from endpoint {1} with timeout {2}ms", bytesToRead, spectralReader, timeoutMS);
+                    logger.debug("readImage: attempting to read {0} bytes of spectrum from endpoint {1} with timeout {2}ms", bytesToRead, spectralReader, timeoutMS);
                     err = spectralReader.Read(subspectrumBytes, bytesReadThisEndpoint, chunk_size, timeoutMS, out bytesRead);
-                    logger.debug("readSubspectrum: read {0} bytes of spectrum from endpoint {1} (ErrorCode {2})", bytesRead, spectralReader, err.ToString());
+                    logger.debug("readImage: read {0} bytes of spectrum from endpoint {1} (ErrorCode {2})", bytesRead, spectralReader, err.ToString());
                 }
                 catch (Exception ex)
                 {
-                    logger.error("readSubspectrum: caught exception reading endpoint: {0}", ex.Message);
-                    return null; //  break;  // should be return null;
+                    logger.error("readImage: caught exception reading endpoint: {0}", ex.Message);
+                    return null; 
                 }
 
                 bytesReadThisEndpoint += bytesRead;
-                logger.debug("readSubspectrum: bytesReadThisEndpoint now {0}", bytesReadThisEndpoint);
+                logger.debug("readImage: bytesReadThisEndpoint now {0}", bytesReadThisEndpoint);
 
                 if (bytesReadThisEndpoint == 0 && !triggerWasExternal)
                 {
-                    logger.error("readSpectrum: read nothing (timeout?)");
-                    return null; //  break; // should be return null
+                    logger.error("readImage: read nothing (timeout?)");

```

Comment:
These changes were necessitated when I found 3 separate functions that could generate "readSpectrum: read nothing (timeout?)"

Reply:
If this is the area scan function...it needs an overhaul anyway once we get the firmware